### DISCUSSION TITLE
targeting a specific folder for search

### DISCUSSION THREAD

**Discussion by pdurbin** [sphinx-doc/sphinx#13057] [Q&A]
Hi! We have a variety of guides at https://guides.dataverse.org with fairly different audiences (user, admin, developer, sysadmin, etc.). It's there any easy way to configure search to target a specific guide?

Perhaps search could even default to the user guide since our biggest audience is users, unsurprisingly. :sweat_smile: Right now the poor users get lots of hits for stuff that's pretty confusing for them as it's meant for developers, sysadmins, etc. :disappointed: Here's our [source](https://github.com/IQSS/dataverse/tree/develop/doc/sphinx-guides) if it's helpful. Each guide is in its own folder like "user", etc. Thanks!

**Comment by liborjelinek** [comment]
IMHO it's not possible, search is always per-project. You could consider split doc into four Sphinx projects and link it with intersphinx extension (to have no dead links).

**Reply by jayaddison** [reply]
Although that could work, one potential problem with dividing the documentation into different projects is that some categories of reader might expect or rely on being able to find documentation from the other segments, and would be confused to find that information absent (in a way, the exact reverse problem).

Another limitation is that it would require some care to keep the styling, configuration and so on consistent across the different projects and their builds.  Not insurmountable, but enough to make me wonder what alternatives might be possible.

@pdurbin I notice in the search functionality for `dataverse.org` itself, there is support for filtering on facets/tags of the datasets.  I wonder whether something similar could be possible by adding optional per-document (?) tags -- or properties such as directory path -- to Sphinx projects.

![image](https://github.com/user-attachments/assets/fdab0f8d-9b89-4957-b858-a18f1cf09005)

**Reply by pdurbin** [reply]
@jayaddison yes, for Dataverse itself we use Solr and take advantage of its faceting feature.

**Reply by jdillard** [reply]
I made an internal extension that searches multiple sphinx projects (that you configure), which would work in the case of splitting up the projects, but it is pretty version dependent as the javascript is fairly heavily modified so I haven't open sourced it.

I'm pretty sure you could make a similar extension where instead of splitting up, you add help text based on that parent folder of the search results (that looks like how you have things segmented). That should require fewer changes and make maintenance a lot easier.

**Reply by pdurbin** [reply]
@jdillard interesting. Thanks. I'm hoping to avoid modifying any javascript but it's good to know that you think it's possible!
